### ISSUE TITLE
Dialyzer case clause error on OTP 25+

### ISSUE THREAD

**Issue by zachdaniel** [erlang/otp#6323]
**Describe the bug**
Getting a case clause error when running dialyzer, on all versions of OTP 25+.

```erlang
:dialyzer.run error: Analysis failed with error:
{{case_clause,[{c_literal,[0,{file,"0"},compiler_generated],'<>'},
               {c_var,[{label,1160433},82,{file,"0"},compiler_generated],
                      '_@1'},
               {c_var,[{label,1160432},82,{file,"0"},compiler_generated],
                      '_@2'}]},
 [{dialyzer_dataflow,is_lc_default_clause,1,
                     [{file,"dialyzer_dataflow.erl"},{line,1349}]},
  {dialyzer_dataflow,clause_error,6,
                     [{file,"dialyzer_dataflow.erl"},{line,1304}]},
  {dialyzer_dataflow,do_clause,7,[{file,"dialyzer_dataflow.erl"},{line,1246}]},
  {dialyzer_dataflow,handle_clauses,9,
                     [{file,"dialyzer_dataflow.erl"},{line,1199}]},
  {dialyzer_dataflow,handle_case,3,
                     [{file,"dialyzer_dataflow.erl"},{line,932}]},
  {dialyzer_dataflow,analyze_loop,1,
                     [{file,"dialyzer_dataflow.erl"},{line,186}]},
  {dialyzer_dataflow,get_warnings,5,
                     [{file,"dialyzer_dataflow.erl"},{line,126}]},
  {dialyzer_succ_typings,collect_warnings,2,
                         [{file,"dialyzer_succ_typings.erl"},{line,189}]}]}
```

**To Reproduce**
Run dialyzer on the project with the original issue. Apoligies, its a large Elixir project, and without any hint as to what file/module might be causing the problem its difficult for me to isolate that cause.

**Expected behavior**
Dialyzer should fail (or at least show the file in which this error happened, perhaps?

**Affected versions**
All versions of OTP 25+

**Additional context**

https://github.com/ash-project/ash/issues/382

**Comment by mikpe** [comment]
This looks like a regression from https://github.com/erlang/otp/commit/e55d8a9b209140c40198f4200f56a88aa3d04eeb . Specifically the function that complains, `is_lc_default_clause/1`, doesn't have a catch-all so it only allows `cerl:clause_pats/1` to return 1- or 2-element lists, while in your example you have a 3-element list of patterns. The previous code this was factored out from did have that default clause. @bjorng?

**Comment by bjorng** [comment]
@mikpe Thanks for pointing out the guilty commit!

Fixed in #6326.
